### PR TITLE
[LDAP] Add flag to disable builtin auth mechanism

### PR DESCRIPTION
**Pull request by molotkov-and**
### Changelog entry 

...

### Changelog category 

* Improvement

### Additional information

...

### CODE REVIEW

File: ydb/core/protos/auth.proto [hunk 1]
```diff
@@ -53,6 +53,7 @@ message TAuthConfig {
     optional bool UseBuiltinDomain = 78 [default = true];
     optional string AccessServiceType = 79 [default = "Yandex_v2"]; // For now the following values are supported: "Yandex_v2", "Nebius_v1"
     optional string CertificateAuthenticationDomain = 80 [default = "cert"];
+    optional bool EnableBuiltinAuthMechanism = 81 [default = true];
```
Comment (adameat) [inline ydb/core/protos/auth.proto:56]: it's called "Login".
we bave builtin tokens, and it's a different thing...

File: ydb/core/tx/schemeshard/schemeshard__operation_alter_login.cpp [hunk 2]
```diff
@@ -15,7 +16,9 @@ class TAlterLogin: public TSubOperationBase {
         NIceDb::TNiceDb db(context.GetTxc().DB); // do not track is there are direct writes happen
         TTabletId ssId = context.SS->SelfTabletId();
         auto result = MakeHolder<TProposeResponse>(OperationId.GetTxId(), ssId);
-        if (Transaction.GetWorkingDir() != context.SS->LoginProvider.Audience) {
+        if (!AppData()->AuthConfig.GetEnableBuiltinAuthMechanism()) {
+            result->SetStatus(NKikimrScheme::StatusPreconditionFailed, "Builtin authentication mechanism has been disabled in the cluster settings");
```
Comment (adameat) [inline ydb/core/tx/schemeshard/schemeshard__operation_alter_login.cpp:20]: Login authentication disabled

File: ydb/services/ydb/ydb_ldap_login_ut.cpp [hunk 3]
```diff
@@ -99,9 +99,9 @@ void InitLdapSettingsWithEmptyBindPassword(NKikimrProto::TLdapAuthentication* ld
 
 class TLoginClientConnection {
 public:
-    TLoginClientConnection(std::function<void(NKikimrProto::TLdapAuthentication*, ui16, TTempFileHandle&)> initLdapSettings)
+    TLoginClientConnection(std::function<void(NKikimrProto::TLdapAuthentication*, ui16, TTempFileHandle&)> initLdapSettings, bool isBuiltinAuthMechanismDisabled = true)
```
Comment (adameat) [inline ydb/services/ydb/ydb_ldap_login_ut.cpp:102]: I wouldn't recommend doing it with bools in the constructor, but it's a UT so it's not that important

**Review by adameat — CHANGES_REQUESTED** [review]

File: ydb/core/protos/auth.proto (hunk 1 above)
Comment (molotkov-and) [reply ydb/core/protos/auth.proto]: done

File: ydb/core/tx/schemeshard/schemeshard__login.cpp [hunk 4]
```diff
@@ -66,13 +67,18 @@ struct TSchemeShard::TTxLogin : TSchemeShard::TRwTxBase {
...
+        if (loginRequest.ExternalAuth || AppData(ctx)->AuthConfig.GetEnableBuiltinAuthMechanism()) {
+            NLogin::TLoginProvider::TLoginUserResponse LoginResponse = Self->LoginProvider.LoginUser(loginRequest);
+            if (LoginResponse.Error) {
+                result->Record.SetError(LoginResponse.Error);
+            }
+            if (LoginResponse.Token) {
+                result->Record.SetToken(LoginResponse.Token);
+            }
+        } else {
+            result->Record.SetError("Builtin authentication mechanism has been disabled in the cluster settings");
```
[the remark this answers is not in the document]
Comment (molotkov-and) [reply ydb/core/tx/schemeshard/schemeshard__login.cpp]: done

File: ydb/core/tx/schemeshard/schemeshard__operation_alter_login.cpp (hunk 2 above)
Comment (molotkov-and) [reply ydb/core/tx/schemeshard/schemeshard__operation_alter_login.cpp]: done

**Review by adameat — APPROVED** [review]

**Pull request merged**